src/compas_fab/robots/semantics.py

```diff
@@ -41,6 +43,37 @@ def __init__(
         self.disabled_collisions = disabled_collisions or set()
         self.group_states = group_states or {}
 
+    @property
+    def data(self):
+        data = {
+            "robot_model": self.robot_model,
+            "groups": self.groups,
+            "main_group_name": self.main_group_name,
+            "passive_joints": self.passive_joints,
+            "end_effectors": self.end_effectors,
+            "disabled_collisions": sorted(self.disabled_collisions),
+            "group_states": self.group_states,
+        }
+        return data
+
+    @data.setter
+    def data(self, data):
+        self.robot_model = data.get("robot_model", None)
+        self.groups = data.get("groups", {})
+        self.main_group_name = data.get("main_group_name", None)
+        self.passive_joints = data.get("passive_joints", [])
+        self.end_effectors = data.get("end_effectors", [])
+        self.disabled_collisions = data.get("disabled_collisions", set())
+        if len(self.disabled_collisions) > 0:
+            self.disabled_collisions = {tuple(pair) for pair in self.disabled_collisions}
+        self.group_states = data.get("group_states", {})
+
+    @classmethod
+    def from_data(cls, data):
+        robot_semantics = cls(None)
+        robot_semantics.data = data
+        return robot_semantics
```

Comment:
@gonzalocasas commented on this. I'd say setting `robot_model=None` in arguments of `__init__` is legit. Then you don't need a custom `from_data`. While I was not a fan, this convention grew on me.

Having said that, the trend in COMPAS 2.0 is to get rid of the `data.setter` acknowledging it's often redundant. So you'd then just have a `data` getter and a `from_data` which feeds the `data` dict to `__init__`.

Example (just for illustration, not sure if this works in in LTS): https://github.com/gramaziokohler/compas_timber/commit/d11d83e268a679b016ad60ef9f5be1f027887b48#diff-5cbdc4545536c895404b80de4fdcc522ec5b95dfb4af73a47e4120888cc22031R42-R48

Reply:
I'm not sure if I get your argument. 

Are you saying that the style is moving towards having default values for all parameters in the `__init__` and then skipping the `def from_data()`?

Probably it would be good to have a more wholistic stylistic clean up separately. The style currently used in compas_fab is a mixed bag of things. I can do that as long as we can have some consensus on what convention we adopt.

Reply:
yea sorry didn't wanna cause confusion. Let's leave it like that.